notebooks/problem-sets/quantum_enigma_problem_set.ipynb

```diff
@@ -0,0 +1,381 @@
+    "\n",
+    "3. 85%\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "4. 100%\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can try this on a real quantum system below (which could take up to ~1 hour to run), but you can also solve the question without needing to run on quantum hardware"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "uses-hardware"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# 1. Create a simple quantum program called a 'quantum circuit'.\n",
+    "from qiskit import QuantumCircuit\n",
+    "\n",
+    "qc = QuantumCircuit(3)\n",
+    "\n",
+    "qc.h(0)\n",
+    "qc.h(2)\n",
+    "qc.cx(0, 1)\n",
+    "qc.barrier(0, 1, 2)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(2)\n",
+    "qc.cx(2, 0)\n",
+    "qc.x(2)\n",
+    "qc.barrier(0, 1, 2)\n",
+    "qc.swap(0, 1)\n",
+    "qc.x(1)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(0)\n",
+    "qc.x(2)\n",
+    "qc.cx(2, 0)\n",
+    "qc.x(2)\n",
+    "\n",
+    "measured_qc = qc.measure_all(inplace=False)\n",
+    "\n",
+    "# 2. Ask IBM Quantum for its least busy device that isn't a simulator.\n",
+    "#    If you're running this example locally, you need to load your\n",
+    "#    account with your IBM Quantum API token\n",
+    "# IBMQ.save_account(token=\"XYZ\")\n",
+    "# IBMQ.load_account()\n",
+    "from qiskit.providers.ibmq import IBMQ, least_busy\n",
+    "provider = IBMQ.get_provider('ibm-q')\n",
+    "device = least_busy(\n",
+    "            provider.backends(\n",
+    "                filters= lambda x: not x.configuration().simulator\n",
+    "            )\n",
+    "        )\n",
+    "print(f'Running on {device.name()}')\n",
+    "\n",
+    "# 3. Convert the program to a form the device can run.\n",
+    "#    This is known as 'transpiling'\n",
+    "from qiskit import transpile\n",
+    "transpiled_qc = transpile(measured_qc, device)\n",
+    "\n",
+    "# 4. Send the program off to IBM Quantum to run on a real device\n",
+    "#    and monitor its status.\n",
+    "from qiskit.tools import job_monitor\n",
+    "job = device.run(transpiled_qc)\n",
+    "job_monitor(job)\n",
+    "\n",
+    "# 5. Plot the results as a histogram.\n",
+    "from qiskit.visualization import plot_histogram\n",
+    "plot_histogram(job.result().get_counts())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Share Feedback\n",
+    "\n",
+    "<script src=\"https://static.airtable.com/js/embed/embed_snippet_v1.js\"></script><iframe class=\"airtable-embed airtable-dynamic-height\" src=\"https://airtable.com/embed/shrDTfq6i3a0lJx4c?backgroundColor=purple\" frameborder=\"0\" onmousewheel=\"\" width=\"100%\" height=\"1481\" style=\"background: transparent; border: 1px solid #ccc;\"></iframe>\n",
```

Comment:
just a comment for consideration... maybe add a basic oading message to the iframe (for instances, when it takes longer than expected to load)... feel free to ignore the suggestion or change the `Loading...` message (FYI, `srcdoc=` can be any HTML markup)

```suggestion
    "<script src=\"https://static.airtable.com/js/embed/embed_snippet_v1.js\"></script><iframe class=\"airtable-embed airtable-dynamic-height\" src=\"https://airtable.com/embed/shrDTfq6i3a0lJx4c?backgroundColor=purple\" frameborder=\"0\" onmousewheel=\"\" width=\"100%\" height=\"1481\" style=\"background: transparent; border: 1px solid #ccc;\" srcdoc=\"Loading...\" onload=\"this.removeAttribute('srcdoc')\"></iframe>\n",
```